cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -423,7 +423,7 @@ func ValidateServiceSubnetSize(subnetStr string, fldPath *field.Path) field.Erro
 }
 
 // ValidatePodSubnetNodeMask validates that the relation between podSubnet and node-masks is correct
-func ValidatePodSubnetNodeMask(subnetStr string, c *kubeadm.ClusterConfiguration, fldPath *field.Path) field.ErrorList {
+func ValidatePodSubnetNodeMask(subnetStr string, c *kubeadm.ClusterConfiguration, isDualStack bool, fldPath *field.Path) field.ErrorList {
```

Comment:
i think we should leave the signature of ValidatePodSubnetNodeMask the way it is.

"dual stack" is a feature gate, and functions like getClusterNodeMask can continue extracting the value from the ClusterConfiguration:
`isDualStack := features.Enabled(c.FeatureGates, features.IPv6DualStack)`

`isIPv6` is not a feature gate, it is a condition that matters to `getClusterNodeMask`

Reply:
Emm, just found `isDualStack` is already calculated in parent function and a similar function `ValidateIPNetFromString ` has already done like this.  😅
https://github.com/kubernetes/kubernetes/blob/43e8405d9c675521eff6e19c9bedbe116eb43d9e/cmd/kubeadm/app/apis/kubeadm/validation/validation.go#L495-L506

Reply:
ok, this seems like a separate refactor that can be done in the future.